txstratum/protocol.py

```diff
@@ -372,8 +372,22 @@ def method_submit(self, params: Any, msgid: JSONRPCId) -> None:
             # to solve the block.
             self.manager.submit_solution(self, job, obj.get_struct_nonce())
             if obj.is_block:
+                self.log.info(
+                    "Block solved: ",
+                    miner_id=self.miner_id,
+                    miner_type=self.miner_type,
+                    hashrate=self.hashrate_ghs,
+                    weight=job.get_weight(),
+                )
                 self.blocks_found += 1
             else:
+                self.log.info(
+                    "Transaction solved: ",
```

Comment:
This can be logged twice for the same transaction, if the miner sends two solutions for it before it receives a new job.

We've seen this happening with the `ccminer` in the wild.

If you're going to use these logs for any additional aggregation and analysis later, would this influence in your analysis?

Reply:
This is fine. The information we want to collect is the miner that is able to solve this weight (and its hashrate in that moment)

Reply:
I guess you could add the tx hash so you can filter out duplicated log entries.

Reply:
Done https://github.com/HathorNetwork/tx-mining-service/pull/115/commits/6a03c3952974bf811598bdc1cbfcdc149ed9584d